### PR TITLE
Add support for autolinks

### PR DESCRIPTION
**Pull request by alexal**
Fixes: #2044.

### CODE REVIEW

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:92]:
```suggestion
	u := fmt.Sprintf("repos/%v/%v/autolinks/%v", owner, repo, id)
```

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:72]:
```suggestion
	u := fmt.Sprintf("repos/%v/%v/autolinks/%v", owner, repo, id)
```

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:91]:
```suggestion
func (s *RepositoriesService) DeleteAutolink(ctx context.Context, owner, repo string, id int64) (*Response, error) {
```

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:71]:
```suggestion
func (s *RepositoriesService) GetAutolink(ctx context.Context, owner, repo string, id int64) (*Autolink, *Response, error) {
```

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:13]:
```suggestion
// AutolinkOptions specifies parameters for RepositoriesService.AddAutolink method.
```

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:14]: Let's please delete this line and duplicate the fields in the `Autolink` struct below.
This will be more consistent with the rest of the repo.

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:20]:
```suggestion
// Autolink represents autolinks to external resources like JIRA issues and Zendesk tickets.
```

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:21]: Let's please delete this line.

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:29]: Please add a blank comment line immediately before this line to separate the GitHub API docs for consistency with the other methods in the repo.

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:53]: Please add a blank comment line immediately before this line to separate the GitHub API docs.

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:70]: Please add a blank comment line immediately before this line to separate the GitHub API docs.

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:90]: Please add a blank comment line immediately before this line to separate the GitHub API docs.

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:60]: Please add a blank line before this one for consistency with the repo.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @alexal ! 
This is looking really good.  Please address the mostly-minor issues.

File: github/repos_autolinks.go
Comment (gmlewis) [reply github/repos_autolinks.go:14]: Line 14 can be deleted.

Comment (alexal) [reply github/repos_autolinks.go]: Done.

File: github/repos_autolinks.go
Comment (gmlewis) [inline github/repos_autolinks.go:20]: If you don't mind, I think I prefer the following instead of these two lines because then, if `Autolink` is used anywhere else, this comment won't become stale and won't need updating:

```
// Autolink represents autolinks to external resources like JIRA issues and Zendesk tickets.
```

Comment (alexal) [reply github/repos_autolinks.go]: Sure, done.

**Review by gmlewis — APPROVED** [review]
Thank you, @alexal !
LGTM.

Awaiting second LGTM before merging.

**Comment by alexal** [comment]
@wesleimp ,

Just wondering whether you had a chance to review the change?

**Comment by alexal** [comment]
@willnorris will you be able to do a second review, please?

**Comment by alexal** [comment]
@dmitshur could you please review the changes?

**Comment by gmlewis** [comment]
@alexal - I try not to bother Will or Dmitri with requests for code reviews, in general.
Lately, @Parker77 has been really helpful with code reviews.  Maybe if he has time, he could check this one out.

**Review by Parker77 — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @Parker77!
Merging.

**Pull request merged**
